manager/controlapi/service.go

```diff
@@ -179,6 +180,9 @@ func (s *Server) UpdateService(ctx context.Context, request *api.UpdateServiceRe
 			return errNetworkUpdateNotSupported
 		}
 
+		if request.Spec != nil && reflect.TypeOf(service.Spec.Mode) != reflect.TypeOf(request.Spec.Mode) {
```

Comment:
Can we document with a comment here about why this is disallowed? I just don't want us to forget that this is possible in future.